provider-middleware/brightspace.go

```diff
@@ -202,6 +219,14 @@ func (srv *BrightspaceService) ImportCourses(db *gorm.DB) error {
 	return nil
 }
 
+func (srv *BrightspaceService) updateTotalProgress(db *gorm.DB, totalProgressMilestones int, externalId string) {
+	if db.Where("provider_platform_id = ? AND external_id = ? AND total_progress_milestones = ?", srv.ProviderPlatformID, externalId, totalProgressMilestones).First(&models.Course{}).Error != nil {
+		if err := db.Table("courses as c").Where("provider_platform_id = ? AND external_id = ?", srv.ProviderPlatformID, externalId).Update("total_progress_milestones", totalProgressMilestones).Error; err != nil {
```

Comment:
On line 223, instead of checking for nil error, can we compare `.RowsAffected == 0` 

and line 224 We can remove unused table alias. `db.Model(&models.Course{})` instead of `db.Table("courses as c")`

Reply:
I modified all similar Error checks to be a RowsAffected check, committed and pushed.  Sorry for taking so long to get this completed.